crates/katana/node/src/lib.rs

```diff
@@ -206,16 +219,22 @@ pub async fn build(mut config: Config) -> Result<Node> {
     // --- build l1 gas oracle
 
     // Check if the user specify a fixed gas price in the dev config.
+    // cases to cover:
+    //      1. Fixed price by user
+    //      2. No fixed price by user and no sampling
+    //      3. Sampling with user input provider url
     let gas_oracle = if let Some(fixed_prices) = config.dev.fixed_gas_prices {
+        // Use fixed gas prices if provided in the configuration
         L1GasOracle::fixed(fixed_prices.gas_price, fixed_prices.data_gas_price)
-    }
-    // TODO: for now we just use the default gas prices, but this should be a proper oracle in the
-    // future that can perform actual sampling.
-    else {
+    } else if config.l1_provider_url.as_ref().is_none() {
+        // Use default fixed gas prices if `gpo` is `None`
         L1GasOracle::fixed(
             GasPrices { eth: DEFAULT_ETH_L1_GAS_PRICE, strk: DEFAULT_STRK_L1_GAS_PRICE },
             GasPrices { eth: DEFAULT_ETH_L1_DATA_GAS_PRICE, strk: DEFAULT_STRK_L1_DATA_GAS_PRICE },
         )
+    } else {
+        // Default to a sampled gas oracle using the given provider
+        L1GasOracle::sampled(config.l1_provider_url.clone())
     };
```

Comment:
```suggestion
    } else if let Some(url) = config.l1_provider_url {
             // Default to a sampled gas oracle using the given provider
        L1GasOracle::sampled(config.l1_provider_url.clone())
    } else {
       // Use default fixed gas prices if `gpo` is `None`
        L1GasOracle::fixed(
            GasPrices { eth: DEFAULT_ETH_L1_GAS_PRICE, strk: DEFAULT_STRK_L1_GAS_PRICE },
            GasPrices { eth: DEFAULT_ETH_L1_DATA_GAS_PRICE, strk: DEFAULT_STRK_L1_DATA_GAS_PRICE },
        )
    };
```

complements the comment about accepting an `Option` is an anti pattern

Reply:
I didn't realize, thank you for pointing that out!